runtime/src/main/java/org/corfudb/protocols/wireprotocol/ClientHandshakeHandler.java

```diff
@@ -69,54 +81,51 @@ public ClientHandshakeHandler(@NonNull UUID clientId, UUID serverId, int handsha
      * @throws Exception
      */
     @Override
-    public void channelRead(ChannelHandlerContext ctx, Object m)
-        throws Exception {
-
+    public void channelRead(ChannelHandlerContext ctx, Object m) throws Exception {
         if (this.handshakeState.failed()) {
-            // if handshake has already failed, return
+            log.warn("channelRead: Handshake has failed. Dropping message - {}", m);
             return;
         }
 
-        if (this.handshakeState.completed()) {
-            // If handshake completed successfully, but still a message came through this handler,
-            // send on to the next handler in order to avoid message loss.
-            super.channelRead(ctx, m);
+        if (m instanceof CorfuMsg) {
+            CorfuMsg msg = (CorfuMsg) m;
+            if (this.handshakeState.completed()) {
+                // Only send upstream if handshake is complete.
```

Comment:
I think the original comment is more clear, can you restore that?

Reply:
Done.